src/Rings/AbelianClosure.jl

```diff
@@ -933,25 +933,22 @@ end
 
 # Construct the map from `F` to an abelian closure `K` such that `gen(F)`
 # is mapped to `x`.
-# If `F` is a cyclotomic field with conductor `N` then assume that `gen(F)`
-# is mapped to `QQAbFieldElem(gen(F), N)`.
+# If `F` has conductor `N` then assume that `x.c == N` holds.
+# If `F` is a cyclotomic field with conductor `N` then assume that
+# `x == QQAbFieldElem(gen(F), N)`.
 # (Use that the powers of this element form a basis of the field.)
 function _embedding(F::QQField, K::QQAbField{AbsSimpleNumField},
                     x::QQAbFieldElem{AbsSimpleNumFieldElem})
-  C1, z = cyclotomic_field(1)
+  C1, _ = cyclotomic_field(1)
 
   f = function(x::QQFieldElem)
     return QQAbFieldElem(C1(x), 1)
   end
 
   finv = function(x::QQAbFieldElem; check::Bool = false)
-    if conductor(x) == 1
-      return Hecke.force_coerce_cyclo(C1, data(x))
-    elseif check
-      return
-    else
-      error("element has no preimage")
-    end
+    res = Hecke.force_coerce_cyclo(C1, data(x), Val(false))
+    !check && res === nothing && error("element has no preimage")
```

Comment:
Hm, if `check == true` and `res === nothing`, will this return nothing? Sorry, I am not good with logic.

Reply:
This code is rather confusing to me as well. But so is the old code: it indeed also returns `nothing` when `check` is `true` and `conductor(x) != 1` ?!

Reply:
I guess what confuses me is that `check` apparently has a different meaning here than it has in e.g. our constructors. Since I don't know what the meaning of `check` is supposed to be here, I can't comment further.

Is it documented somewhere?

Reply:
@fingolfin See my comment below: The function in question is internal and not documented (as well as the functions called by it). I think we could change the meaning of `check` to that in other situations.